### PR TITLE
Use the metrics collector in the UsageTracker

### PR DESCRIPTION
**Pull request by tempoz**
Change UsageTracker to use MetricsCollector in preparation for flag changes.

This means that registration of the UsageTracker will no longer require access
to the bare `rbuf` (`redisutil.CommandBuffer`), and can instead depend solely
on the `env`.

---

**Version bump**: Minor

### CODE REVIEW

File: server/util/testing/flags/flags.go [hunk 1]
```diff
@@ -7,10 +7,21 @@ import (
...
 
 	"github.com/buildbuddy-io/buildbuddy/server/config"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+// Set a flag value and register a cleanup function to restore the flag
+// to its original value after the given test is complete.
+func SetAndReconcile(t testing.TB, name, value string, env environment.Env) {
```
Comment (tylerwilliams) [inline server/util/testing/flags/flags.go:17]: nit: add a TODO that this will be removed in final cleanup?

**Review by tylerwilliams — APPROVED** [review]
LGTM as long as @bduffany is ok with it

File: server/util/testing/flags/flags.go (hunk 1 above)
Comment (tempoz) [reply server/util/testing/flags/flags.go:17]:
```suggestion
// TODO: Remove this function when we no longer need to reconcile flags
func SetAndReconcile(t testing.TB, name, value string, env environment.Env) {
```

File: server/backends/memory_metrics_collector/memory_metrics_collector.go [hunk 2]
```diff
@@ -32,6 +33,35 @@ func NewMemoryMetricsCollector() (*MemoryMetricsCollector, error) {
 	}, nil
 }
 
+func (m *MemoryMetricsCollector) IncrementCounts(ctx context.Context, key string, counts map[string]int64) error {
```
Comment (bduffany) [inline server/backends/memory_metrics_collector/memory_metrics_collector.go:36]: mind adding unit tests for IncrementCounts and IncrementCountsWithExpiry? We don't have any use cases that would directly exercise these new codepaths so it'd be nice to have some coverage

**Review by bduffany — CHANGES_REQUESTED** [review]

File: server/backends/memory_metrics_collector/memory_metrics_collector.go (hunk 2 above)
Comment (tempoz) [reply server/backends/memory_metrics_collector/memory_metrics_collector.go:36]: Sure, sounds good.

**Pull request merged**